### ISSUE TITLE
Avail tests should clear repos (or suggest it, if run as part of build)

### ISSUE THREAD

**Issue by tdhsmith** [AvailLang/Avail#162]
If your repositories get to a broken state, the build process will error out because the test suite will fail (in my case failing`testLoadStandardLibraries` because no module could be loaded). These errors give no clue that the error may not be coming from the source files themselves.

My thought is that the test suite should always clear the repos, so that running it is as close to stateless as possible (and repos seem to be considered _state_ rather than _source_), however that does introduce a speed hit for the core team and there might be utility to running the default tests against dirty repos that I'm not aware of. If either of those seem like dealbreakers I'd acquiesce to either:
-  only enforcing it during a build, or
- displaying a message about possible repo issues when tests fail and an optional build flag/command to clear them

Happy to handle myself, just wanted input.

**Comment by tdhsmith** [comment]
Err, pause a moment on this analysis. I might have mischaracterized the whole thing, because I was clearing the wrong repos anyway. I don't get how it could've fixed itself though. Hrm.

**Comment by markATavail** [comment]
Yeah, the "Appropriate register for kind not found" problem popped up when Leslie exposed a deep optimizer bug, probably by having two blocks that referenced a label in an outer scope.  It's been fixed for a few weeks (on the development branch), if my sense of time can be believed these days.
